third_party/move/tools/move-package/src/compilation/compiled_package.rs

```diff
@@ -747,13 +747,28 @@ impl CompiledPackage {
             if skip_attribute_checks {
                 flags = flags.set_skip_attribute_checks(true)
             }
-            let model = run_model_builder_with_options_and_compilation_flags(
-                vec![sources_package_paths],
-                deps_package_paths.into_iter().map(|(p, _)| p).collect_vec(),
-                ModelBuilderOptions::default(),
-                flags,
-                &known_attributes,
-            )?;
+            let model = if resolution_graph.build_options.generate_docs
```

Comment:
Still use V1 generated model for docgen and abigen

Reply:
Why is this necessary?

Reply:
When V2 model is used for docgen, links generated for modules from other packages are not correct because they are treated as targets. See the code [here](https://github.com/aptos-labs/aptos-core/blob/a72ef8a716ecf3ab207c8377cb94c9c5aedaf5b4/third_party/move/move-prover/move-docgen/src/docgen.rs#L470). abigen is also excluded in case I miss anything.

Reply:
Please add appropriate comments here explaining this conditional check.